src/storage/mod.rs

```diff
@@ -39,6 +40,51 @@ pub enum KeyedWriteError<E> {
     Store(#[from] E),
 }
 
+#[pin_project]
+#[derive(Debug)]
+pub struct Content<R>(u64, #[pin] R);
```

Comment:
Is it necessary for this to be a tuple struct? I think a named struct would make it clearer that the first element is the size.

Reply:
fair enough, everything gets compiled to the same size anyway 👍